src/app/migrations/1.21.0.js

```diff
@@ -0,0 +1,98 @@
+const debug = require('debug')('mongodb-compass:migrations');
+const asyncr = require('async');
+const fs = require('fs');
+const path = require('path');
+const { remote } = require('electron');
+
+/**
+ * The plugin database name.
+ */
+const INDEXED_DB = 'compass-aggregations';
+
+/**
+ * The collection name.
+ */
+const PIPELINES = 'pipelines';
+
+/**
+ * Saved pipelines on disk.
+ */
+const FS_DIR = 'SavedPipelines';
+
+/**
+ * Upgrade the db.
+ *
+ * @param {IDBDatabase} db - The database.
+ */
+const upgradeDb = (db) => {
+  const store = db.createObjectStore(PIPELINES);
+  store.createIndex('namespace', 'namespace', { unique: false });
+};
+
+/**
+ * Get the object store for the provided mode.
+ *
+ * @param {String} mode - The mode.
+ * @param {Function} done - The callback.
+ */
+const getObjectStore = (mode, done) => {
+  const request = window.indexedDB.open(INDEXED_DB, 1);
+  request.onsuccess = (evt) => {
+    const db = evt.target.result;
+    const transaction = db.transaction(PIPELINES, mode);
+    done(transaction.objectStore(PIPELINES));
+  };
+
+  request.onupgradeneeded = (evt) => {
+    upgradeDb(evt.target.result);
+  };
+};
+
+/**
+ * This migration removes aggregations from IndexedDB and persists them to
+ * disk.
+ *
+ * @param {Function} done - The done callback.
+ */
+const moveToDiskStorage = (done) => {
+  debug('migration pipelines: moveToDiskStorage');
+  const userDataDir = remote.app.getPath('userData');
+  getObjectStore('readwrite', (store) => {
+    const request = store.getAll();
+    request.onsuccess = (evt) => {
+      const savedPipelines = evt.target.result;
+      if (savedPipelines) {
```

Comment:
Can't we just return done() if !savedPipelines || savedPipelines.length === 0?

Reply:
I'd still want it to create the directory and it wouldn't with that additional check.